### PR TITLE
Support Jenkins 2.440.x

### PR DESCRIPTION
**Pull request by cyrille-leclerc**
Support Jenkins 2.440.x to align with the https://github.com/jenkinsci/junit-sql-storage-plugin so we can integrate it with the the openelemetry-api-plugin 

### Testing done

yes, creating branches of the junit-sql-storage-plugin and opentelemetry-plugin

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Alternative: https://github.com/jenkinsci/junit-sql-storage-plugin/pull/417

File: pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
         <opentelemetry-instrumentation.version>2.5.0</opentelemetry-instrumentation.version>
         <opentelemetry-semconv>1.25.0-alpha</opentelemetry-semconv>
 
-        <jenkins.version>2.440.3</jenkins.version>
+        <jenkins.version>2.414.3</jenkins.version>
```
Comment (timja) [inline pom.xml:31]: we can do https://github.com/jenkinsci/junit-sql-storage-plugin/pull/417 instead

Comment (cyrille-leclerc) [reply pom.xml:31]: I saw that the Jenkins OTel Plugin stil has a lot of users on 2.414.x so downgrading makes sense to me. I'll defer to you.

**Comment by timja** [comment]
see https://github.com/jenkinsci/junit-sql-storage-plugin/releases/tag/324.v90e2a_a_a_a_0dd7

**Comment by cyrille-leclerc** [comment]
Let's stick to `2.440.3`

**Comment by cyrille-leclerc** [comment]
FYI we are releasing 
https://github.com/jenkinsci/opentelemetry-api-plugin/actions/runs/9563166601

**Pull request closed without merging**